### PR TITLE
When using Redis, clear the alloptions cache whenever an option is added/updated/deleted

### PR DESCRIPTION
**Pull request by nathanielks**
Fixes https://github.com/humanmade/hm-platform/issues/132.

### CODE REVIEW

File: inc/redis_alloptions/namespace.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	add_action( 'deleted_option', __NAMESPACE__ . '\maybe_clear_alloptions_cache' );
+}
+
+/**
+ * Fix a race condition in alloptions caching
+ *
+ * See https://core.trac.wordpress.org/ticket/31245#comment:57
+ */
+function maybe_clear_alloptions_cache( $option ) {
+	if ( ! wp_installing() ) {
```
Comment (joehoyle) [inline inc/redis_alloptions/namespace.php]: I'd return early, usually the cleaner pattern we use.

File: inc/redis_alloptions/namespace.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace HM\Platform\Redis_Alloptions;
+
+function bootstrap() {
+	add_action( 'added_option', __NAMESPACE__ . '\maybe_clear_alloptions_cache' );
```
Comment (joehoyle) [inline inc/redis_alloptions/namespace.php]: Elsewhere we usually escape `\` as `\\`, i.e. ` __NAMESPACE__ . '\\maybe_clear_alloptions_cache`

File: inc/redis_alloptions/namespace.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace HM\Platform\Redis_Alloptions;
```
Comment (joehoyle) [inline inc/redis_alloptions/namespace.php]: I'm thinking we don't need to actually call this namespace "redis", just only load it when using redis. maybe `Alloptions_Fix` or something?

**Comment by nathanielks** [comment]
@joehoyle updated!

**Pull request merged**